File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[200];

void dfs(int cur, int n, const vector<vector<int>>& computers) {
    visited[cur] = true;
    for(int nxt = 0; nxt < n; nxt++) {
        if(computers[cur][nxt] == 1 && !visited[nxt]) {
            dfs(nxt, n, computers);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int ans = 0;
    
    for (int i = 0; i < n; i++) {
        if (!visited[i]) {
            dfs(i, n, computers);
            ans ++;
        }
    }
    
    return ans;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    in_net = set()
    
    for i in range(len(computers)):
        if i not in in_net: # 그룹 찾아가자~~
            answer += 1
            
            q = deque([i])
            
            while q:
                cur = q.popleft()
                in_net.add(cur)
                
                for idx, check in enumerate(computers[cur]):
                    if check and idx not in in_net:
                        q.append(idx)
    
    return answer